lib/fuel-rest.js

```diff
@@ -106,6 +106,7 @@ FuelRest.prototype.apiRequest = function(options, callback) {
 
 					// check if we should retry req
 					if(helpers.isValid401(res) && retry) {
+						self.AuthClient.invalidateToken();
```

Comment:
This needs to be `this.AuthClient.invalidateToken(accessToken);` and above this, we need to do `var accessToken = authResponse.accessToken` before the callback and also only if `options.headers.Authorization` was not provided.

@nathan-boyd